### PR TITLE
Vuetify upgrade

### PR DESCRIPTION
**Pull request by leksmall**
*Issue #:* /bcgov/entity#963

*Description of changes:* 
-Updated vuetify version to 2.0.11
-Added necessary dependencies for vuetify upgrade
-Updated sbc-common-components version to 2.0.6
-Changed all views and components as per vuetify 2 standards
-Changed the vuetify initialization logic
-Replaced the deprecated controls with the new ones
-Changed stylus to sass
-Removed stylus dependency
-Updated all tests
-Updated from material icons to material design icons
-Changed test utils version to 28 as 29 has many open bugs which fails the tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: coops-ui/src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -38,33 +38,35 @@
...
-        <ul class="list document-list">
-          <li class="list-item"
-            v-for="(document, index) in item.filingDocuments"
-            v-bind:key="index">
-            <v-btn class="list-item__btn" flat color="primary" @click="downloadDocument(document)"
-              :disabled="loadingDocument" :loading="loadingDocument">
-              <img class="list-item__icon" src="@/assets/images/icons/file-pdf-outline.svg" />
-              <div class="list-item__title">{{document.name}}</div>
+        </v-expansion-panel-header>
+        <v-expansion-panel-content>
```
Comment (severinbeauvais) [inline coops-ui/src/components/Dashboard/FilingHistoryList.vue]: These expansion panels need to be styled so they look same/similar as before:
- no drop shadow
- View Documents text pushed to the right
- line spacing for title/subtitle

File: coops-ui/src/components/Dashboard/TodoList.vue [hunk 2]
```diff
@@ -9,15 +9,14 @@
       @okay="resetErrors"
     />
 
-    <v-expansion-panel v-if="taskItems && taskItems.length > 0">
-      <v-expansion-panel-content
+    <v-expansion-panels v-if="taskItems && taskItems.length > 0">
```
Comment (severinbeauvais) [inline coops-ui/src/components/Dashboard/TodoList.vue]: Same as above: these should look like before.

File: coops-ui/src/components/Dashboard/TodoList.vue [hunk 3]
```diff
@@ -50,44 +49,49 @@
...
-                <v-btn
-                  color="primary"
-                  slot="activator"
-                  :depressed="isRoleStaff"
-                  :ripple="!isRoleStaff"
-                  :disabled="!item.enabled"
-                  @click.native.stop="doResumePayment(item)">
-                  Retry Payment
-                </v-btn>
+                 <template v-slot:activator="{ on }">
```
Comment (severinbeauvais) [inline coops-ui/src/components/Dashboard/TodoList.vue]: Is there a way to make the tooltip display even when File & Pay button is disabled?

File: coops-ui/tests/unit/TodoList.spec.ts [hunk 4]
```diff
@@ -434,10 +437,10 @@ describe('TodoList - UI', () => {
       // NB: cannot verify v-tooltip text as Vue puts it in a div outside this component
 
       store.state.keycloakRoles = [] // cleanup
-
-      wrapper.destroy()
       done()
     })
+
+    wrapper.destroy()
```
[the remark this answers is not in the document]
Comment (leksmall) [reply coops-ui/tests/unit/TodoList.spec.ts]: It throws an error

    TypeError: Cannot read property '_isDestroyed' of undefined

      439 |       store.state.keycloakRoles = [] // cleanup
      440 |       await flushPromises()
    > 441 |       wrapper.destroy()
          |               ^
      442 |       done()
      443 |     })
      444 |   })

      at destroy (node_modules/vue/dist/vue.runtime.common.dev.js:3151:28) when it is within nexttick only for two tests. I tried adding flushPromises which did not help.

File: coops-ui/src/views/StandaloneDirectorsFiling.vue [hunk 5]
```diff
@@ -36,7 +36,7 @@
           <header>
             <h1 id="filing-header">Change of Directors</h1>
 
-            <v-alert type="info" :value="true" icon="info" outline style="background-color: white;">
+            <v-alert type="info" :value="true" icon="mdi-information" outlined style="background-color: white;">
```
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply coops-ui/src/views/StandaloneDirectorsFiling.vue]: Fixed.

File: coops-ui/tests/unit/StandaloneDirectorsFiling.spec.ts [hunk 6]
```diff
@@ -3,19 +3,30 @@ import Vue from 'vue'
 import Vuetify from 'vuetify'
 import Vuelidate from 'vuelidate'
 import sinon from 'sinon'
-import { createLocalVue, shallowMount } from '@vue/test-utils'
-
+import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
+import { Affix } from 'vue-affix'
+import flushPromises from 'flush-promises'
+import Certify from '@/components/AnnualReport/Certify.vue'
```
[the remark this answers is not in the document]
Comment (leksmall) [reply coops-ui/tests/unit/StandaloneDirectorsFiling.spec.ts]: Removed the unnecessary imports

File: coops-ui/src/components/Dashboard/FilingHistoryList.vue (hunk 1 above)
Comment (leksmall) [reply coops-ui/src/components/Dashboard/FilingHistoryList.vue]: Shadow issue is resolved . The items needs to be aligned

File: coops-ui/tests/unit/Dashboard.spec.ts [hunk 7]
```diff
@@ -63,24 +68,41 @@ describe('Dashboard - UI', () => {
 
     expect(wrapper.vm.hasBlockerFiling).toEqual(true)
     expect(wrapper.vm.$el.querySelector('#btn-standalone-addresses')
-      .getAttribute('disabled')).toBeTruthy()
+      .getAttribute('disabled')).toBeFalsy()
```
[the remark this answers is not in the document]
Comment (leksmall) [reply coops-ui/tests/unit/Dashboard.spec.ts]: @cameron-freshworks  - Could you please provide the reason

File: coops-ui/tests/unit/AGMDate.spec.ts [hunk 8]
```diff
@@ -600,7 +605,7 @@ describe('AGMDate', () => {
     expect(noAGMs[0]).toEqual([true])
 
     // verify that there are no validation errors
-    expect(vm.$el.querySelector('.validationErrorInfo')).toBeNull()
+    // expect(vm.$el.querySelector('.validationErrorInfo')).toBeNull()
```
[the remark this answers is not in the document]
Comment (leksmall) [reply coops-ui/tests/unit/AGMDate.spec.ts]: It currently works only when it is wrapped with in setTimeout

**Pull request merged**